graph/multigraph.go

```diff
@@ -0,0 +1,163 @@
+// Copyright ©2014 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package graph
+
+// Line is an edge in a multigraph. A Line returns an ID that must
+// distinguish Lines sharing Node end points.
+type Line interface {
+	Edge
+	ID() int64
+}
+
+// WeightedLine is a weighted multigraph edge.
+type WeightedLine interface {
+	Line
+	Weight() float64
+}
+
+// Multigraph is a generalized multigraph.
+type Multigraph interface {
+	// Has returns whether the node exists within the multigraph.
+	Has(Node) bool
+
+	// Nodes returns all the nodes in the multigraph.
+	Nodes() []Node
+
+	// From returns all nodes that can be reached directly
+	// from the given node.
+	From(Node) []Node
+
+	// HasEdgeBetween returns whether an edge exists between
+	// nodes x and y without considering direction.
+	HasEdgeBetween(x, y Node) bool
+
+	// Lines returns the lines from u to v if any such lines
+	// exist and nil otherwise. The node v must be directly
+	// reachable from u as defined by the From method.
+	Lines(u, v Node) []Line
+}
+
+// WeightedMultigraph is a weighted multigraph.
+type WeightedMultigraph interface {
+	Multigraph
+
+	// WeightedEdge returns the weighted edge from u to v if
```

Comment:
WeightedLines returns the weighted lines from u to v if any such lines exist ...

Reply:
Done